### PR TITLE
Refactored Codeception tests

### PR DESCRIPTION
**Pull request by DavertMik**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | no
| Breaks BC?    | yes
| Tests pass?   | yes
| Fixed issues  | no

Codeception tests are refactored. 

* Structure changed. Each application now contains its own set of tests located in `tests` fir. So there are:
    * `frontend/tests`
    * `common/tests`
    * `backend/tests`
* Test configs moved to `config` folder of corresponding app. Configs simplified
* `codeception/specify` removed from dependencies
* `yiisoft/yii-codeception` removed from dependencies
* `codeception/base` added to dev dependencies
* dependencies marked as stable
* backend/acceptance tests removed
* frontend/acceptance tests disabled and simplified

Should be merged after Codeception 2.2.3 is released

### CODE REVIEW

**Comment by agungsijawir** [comment]
Amazing!

**Comment by samdark** [comment]
It's green! Yay!

File: composer.json [hunk 1]
```diff
@@ -12,15 +12,16 @@
         "irc": "irc://irc.freenode.net/yii",
         "source": "https://github.com/yiisoft/yii2"
     },
-    "minimum-stability": "dev",
+    "minimum-stability": "stable",
```
Comment (samdark) [inline composer.json]: Should be left as `dev`.

File: composer.json [hunk 2]
```diff
@@ -12,15 +12,16 @@
...
     "require": {
         "php": ">=5.4.0",
         "yiisoft/yii2": ">=2.0.6",
         "yiisoft/yii2-bootstrap": "*",
         "yiisoft/yii2-swiftmailer": "*"
     },
     "require-dev": {
-        "yiisoft/yii2-codeception": "*",
+        "codeception/base": "^2.2.3",
+        "codeception/verify": "*",
```
Comment (samdark) [inline composer.json]: Constraint should be specified.

File: frontend/tests/functional/ContactCest.php [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        $I->seeValidationError('Name cannot be blank');
+        $I->seeValidationError('Email cannot be blank');
+        $I->seeValidationError('Subject cannot be blank');
+        $I->seeValidationError('Body cannot be blank');
+        $I->seeValidationError('The verification code is incorrect');
+    }
+
+    public function checkContactSubmitNotCorrectEmail(FunctionalTester $I)
+    {
+        $I->submitForm('#contact-form', [
```
Comment (samdark) [inline frontend/tests/functional/ContactCest.php]: Formatting should be fixed.

File: frontend/tests/functional/ContactCest.php [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        $I->dontSeeValidationError('Name cannot be blank');
+        $I->dontSeeValidationError('Subject cannot be blank');
+        $I->dontSeeValidationError('Body cannot be blank');
+        $I->dontSeeValidationError('The verification code is incorrect');
+    }
+
+    public function checkContactSubmitCorrectData(FunctionalTester $I)
+    {
+        $I->submitForm('#contact-form', [
+            'ContactForm[name]'			=>	'tester',
```
Comment (samdark) [inline frontend/tests/functional/ContactCest.php]: Formatting should be fixed,

**Comment by PommeVerte** [comment]
Awesome. How would one upgrade to this on existing projects though?

**Comment by samdark** [comment]
@PommeVerte manually. These are application boilerplates so once used it's all yours.

**Comment by rob006** [comment]
@PommeVerte 
> Awesome. How would one upgrade to this on existing projects though?

You could try apply a patch: https://patch-diff.githubusercontent.com/raw/yiisoft/yii2-app-advanced/pull/190.patch

**Comment by PommeVerte** [comment]
@samdark @rob006 Thanks guys. I'll wait for this to hit release and then I'll see how to manually make the change.

**Comment by tunecino** [comment]
@rob006 that would be a nice tip to mention in DOCS or as a [cookbook](https://github.com/samdark/yii2-cookbook) article.

**Comment by DavertMik** [comment]
Btw, we already have docs for Yii on codeception site (mentioning advanced application as well) 

Check it out
http://codeception.com/for/yii

File: composer.json (hunk 2 above)
Comment (DavertMik) [inline composer.json]: It is not 1.0 yet, and no breaking changes are planned, so I don't know hot better to set version constraints.

**Comment by DavertMik** [comment]
Added some documentation

File: composer.json (hunk 2 above)
Comment (samdark) [inline composer.json]: ~0.3.1?

**Comment by samdark** [comment]
Merged. Thank you @DavertMik, @Dell-, @DumpOfTheVar, @dis1!

**Comment by chervand** [comment]
What's the reason of removing `codeception/specify` dependency? Was it done only for template simplification purposes or there is something we should consider about?

**Comment by samdark** [comment]
Only for template simplification.

**Pull request closed without merging**